### PR TITLE
Added the shade OSP library and added helpers for customer db automation

### PR DESCRIPTION
**Pull request by san7ket**

### CODE REVIEW

File: upgrade/helpers/tasks.py [hunk 1]
```diff
@@ -481,3 +483,14 @@ def upgrade_using_foreman_maintain():
...
             os.environ.get('TO_VERSION')))
+
+
+def get_osp_hostname(ipaddr):
+    """The openstack has floating ip and we need to fetch the hostname from DNS
+    :param ipaddr : IP address of the osp box
+    """
+    try:
+        result = socket.gethostbyaddr(ipaddr)[0]
+        return result
```
Comment (ntkathole) [inline upgrade/helpers/tasks.py]: extra variable

File: upgrade/helpers/openstack.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import yaml
 import sys
-import time
 
 from fabric.api import env, execute, run
-from novaclient.client import Client
+
 from upgrade.helpers.logger import logger
-from upgrade.helpers.tools import get_hostname_from_ip, host_pings
+from upgrade.helpers.tools import host_pings, host_ssh_availability_check
```
Comment (ntkathole) [inline upgrade/helpers/openstack.py]: nitpick. ascending order

File: upgrade/helpers/openstack.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+
 import os
+import shade
+import yaml
 import sys
-import time
 
 from fabric.api import env, execute, run
-from novaclient.client import Client
+
```
Comment (jyejare) [inline upgrade/helpers/openstack.py]: Remove this line

File: upgrade/helpers/openstack.py [hunk 4]
```diff
@@ -1,12 +1,14 @@
-import novaclient
+
 import os
+import shade
+import yaml
```
Comment (jyejare) [inline upgrade/helpers/openstack.py]: Ascendinfg order

File: upgrade/helpers/openstack.py [hunk 5]
```diff
@@ -36,98 +38,105 @@ def get_openstack_client():
...
-    image = openstack_client.images.find(name=image_name)
-    flavor = openstack_client.flavors.find(name=flavor_name)
-    network = openstack_client.networks.find(label=network_name)
-    floating_ip = openstack_client.floating_ips.create(
-        openstack_client.floating_ip_pools.list()[0].name
-    )
+    env.user = 'root'
+    env.disable_known_hosts = True
+    if flavor_name is None:
+        flavor_name = os.environ.get('FLAVOR_NAME')
```
Comment (jyejare) [inline upgrade/helpers/openstack.py]: ```flavor_name = flavor_name or os.environ.get('FLAVOR_NAME')```, Do this for all.

File: upgrade/helpers/openstack.py [hunk 6]
```diff
@@ -36,98 +38,105 @@ def get_openstack_client():
...
+    instance = openstack_client.create_server(
         name=instance_name,
         image=image.id,
-        flavor=flavor.id,
+        flavor=flavor_name,
         key_name=ssh_key,
-        network=network.id
+        network=network_name,
+        volume_size=volume_size,
+        wait=True
```
Comment (jyejare) [inline upgrade/helpers/openstack.py]: Wait .... for How long ?

What is wait here ?

File: upgrade/helpers/openstack.py [hunk 7]
```diff
@@ -36,98 +38,105 @@ def get_openstack_client():
...
-    hostname = get_hostname_from_ip(str(floating_ip.ip))
+    hostname = get_osp_hostname(ip_addr)
     if not hostname:
         sys.exit(1)
-    env['{0}_host'.format(product)] = hostname
     logger.info('Pinging the Hostname:{0} ..........'.format(hostname))
     host_pings(hostname)
     logger.info('SUCCESS !! The obtained hostname from IP is pinged !!')
     # Update the /etc/hosts file
+    execute(lambda: run("hostnamectl set-hostname {0}".format(
```
Comment (jyejare) [inline upgrade/helpers/openstack.py]: You are setting the hostname on an instance, but before that, you are running ```execute``` on a hostname. Do you think that will work? Have you tested it ?

Comment (jyejare) [reply upgrade/helpers/openstack.py]: Its bit confusing, Please explain more about it !

Comment (san7ket) [reply upgrade/helpers/openstack.py]: I am setting the hostname, just to be in sync with DNS not actually required but good to have. :)
Yes, it is tested

File: upgrade/helpers/openstack.py (hunk 6 above)
Comment (san7ket) [reply upgrade/helpers/openstack.py]: It has a Timeout of 180 seconds, if the Floating Ip is not assigned

File: upgrade/helpers/openstack.py [hunk 8]
```diff
@@ -36,98 +38,105 @@ def get_openstack_client():
...
-    )
+    env.user = 'root'
+    env.disable_known_hosts = True
+    if flavor_name is None:
+        flavor_name = os.environ.get('FLAVOR_NAME')
+    if network_name is None:
+        network_name = os.environ.get('NETWORK_NAME')
+    if ssh_key is None:
+        ssh_key = os.environ.get('OSP_SSHKEY')
+    openstack_client = shade.openstack_cloud(cloud='satellite-jenkins')
```
Comment (san7ket) [inline upgrade/helpers/openstack.py]: @jyejare Clouds.yaml , is the default file that the shade object looks at while creating a interface with OSP

**Comment by san7ket** [comment]
instance.info is needed for the sat-clone tool to fetch the hostname of the created OSP instance. Basically needed for Cust DB upgrade

**Comment by san7ket** [comment]
@jyejare Are we good to merge ?

**Comment by jyejare** [comment]
@san7ket , Yes I m good with it.

**Comment by san7ket** [comment]
Tested Job /job/satellite6_db_upgrade_migrate/330/

**Comment by san7ket** [comment]
Wait for merge until https://github.com/SatelliteQE/robottelo-ci/pull/1104

**Pull request merged**